pkg/network/ebpf/c/runtime/http.c

```diff
@@ -1,68 +1,67 @@
+#include <linux/kconfig.h>
 #include "tracer.h"
 #include "bpf_helpers.h"
 #include "ip.h"
 #include "ipv6.h"
-#include "http.h"
-#include "http-buffer.h"
 #include "sockfd.h"
 #include "conn-tuple.h"
 #include "tags-types.h"
+#include "port_range.h"
+#include "http.h"
+#include "https.h"
+
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(4, 6, 0)
+#error "http runtime compilation is only supported for kernel >= 4.6"
+#endif
 
-// TODO: Replace those by injected constants based on system configuration
-// once we have port range detection merged into the codebase.
-#define EPHEMERAL_RANGE_BEG 32768
-#define EPHEMERAL_RANGE_END 60999
 #define HTTPS_PORT 443
 #define SO_SUFFIX_SIZE 3
 
-static __always_inline int is_ephemeral_port(u16 port) {
-    return port >= EPHEMERAL_RANGE_BEG && port <= EPHEMERAL_RANGE_END;
-}
-
-static __always_inline void read_skb_data(struct __sk_buff* skb, u32 offset, char *buffer) {
-    if (skb->len - offset < HTTP_BUFFER_SIZE) {
-        return;
-    }
+static __always_inline void read_into_buffer_skb(char *buffer, struct __sk_buff* skb, skb_info_t *info) {
+    u64 offset = (u64)info->data_off;
 
 #pragma unroll
     for (int i = 0; i < HTTP_BUFFER_SIZE; i++) {
```

Comment:
```suggestion
    for (int i = 0, u64 offset = (u64)info->data_off; i < HTTP_BUFFER_SIZE; i++, offset++) {
```

Reply:
I'm not sure if that works for unrolled loops. Let me check.

Reply:
Yeah clang rejects it 😬